doc/nrf/libraries/modem/location.rst

```diff
@@ -0,0 +1,198 @@
+.. _lib_location:
+
+Location
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+Location library provides functionality for retrieving device's location utilizing different positioning methods
+such as GNSS satellite positioning including Assisted GPS (A-GPS) and predicted GPS (P-GPS) data, cellular positioning or WiFi positioning.
+
+Overview
+********
+
+Location library provides a convenient API for clients to request devices location.
+The client can determine the preference order of the positioning methods to be used along with some other configuration information.
+Once a method fails to provide location, Location library will perform fallback to the next preferred method.
+
+Supported positioning methods are GNSS, cellular and WiFi positioning.
+Both cellular and WiFi positioning take advantage of the base stations seen with corresponding technology, and then utilizing web services for retrieving the location.
+GNSS positioning uses satellites to compute the location of the device. Location library can utilize assistance data (A-GPS and P-GPS) to find the satellites faster.
+
+Implementation
+==============
+
+Location library has a compact API and then location core that handles location method independent part of the functionality such as fallback to the next preferred method and timeouts.
+Each location method has its own implementation for the location retrieval:
+
+* GNSS positioning
+   * :ref:`gnss_interface` for getting satellite information
+   * A-GPS/P-GPS are managed either with
+      * :ref:`lib_nrf_cloud_agps` and :ref:`lib_nrf_cloud_pgps`, or
+      * the application utilizes some other source for the data and uses :c:func:`location_agps_data_process` to pass the data to Location library
+   * A-GPS/P-GPS data format must be as received from :ref:`lib_nrf_cloud_agps`
```

Comment:
This line provides details to the line above, so it could be indented two times to right. Also, the external assistance can only be A-GPS, not P-GPS because of the nRF P-GPS library limitations.

Reply:
Pekka moved the bullets to the same level so I'm not doing anything for this.